pkg/controller/tenant/tenant_controller.go

```diff
@@ -266,23 +275,30 @@ func (tc *TenantController) createDefaultNetworkObject(tenantName string) (error
 	} else {
 		klog.Infof("creating the default network in tenant %q", tenantName)
 		defaultNetwork := arktosv1.Network{}
-		if err := tc.getDefaultNetwork(tenantName, &defaultNetwork); err != nil {
+		if err := tc.constructDefaultNetwork(tenantName, &defaultNetwork); err != nil {
 			failures = append(failures, err)
 		} else {
-			if _, err = tc.networkClient.ArktosV1().NetworksWithMultiTenancy(tenantName).Create(&defaultNetwork); err != nil && !errors.IsAlreadyExists(err) {
+			switch _, err = tc.networkLister.NetworksWithMultiTenancy(tenantName).Get(defaultNetwork.Name); {
```

Comment:
is the default network for a tenant immutable? i.e. if a tenant want to change their default network settings, will this be done by the the TC or other ways ?

Reply:
I think this was a preliminary design. It was never used before until mizar was introduced. This is creating default network solution for tenant. For now, kubectl can be used to update the default network solution for selected tenant(s). As long as the name is not changes, the same network object will be used for tenant default network solution.